chat-plugins/room-events.js

```diff
@@ -9,61 +9,87 @@
+
+let events = {};
+try {
+	events = require(EVENTS_FILE);
+} catch (e) {
+	if (e.code !== 'MODULE_NOT_FOUND') throw e;
+}
+if (!events || typeof events !== 'object') events = {};
+
+function saveEvents() {
+	fs.writeFileSync(EVENTS_FILE, JSON.stringify(events));
+}
+
+
 exports.commands = {
 	events: 'roomevents',
 	roomevent: 'roomevents',
 	roomevents: {
 		'': function (target, room, user) {
-			if (room.battle) return this.errorReply("This command is not meant for battle rooms.");
-			if (!room.events || Object.keys(room.events).length === 0) {
+			if (!room.chatRoomData) return this.errorReply("This command is unavailable in temporary rooms.");
+			if (!events[room] || !Object.keys(events[room]).length) {
 				return this.errorReply("There are currently no planned upcoming events for this room.");
 			}
 			if (!this.runBroadcast()) return;
-			let buff = '<table border="1" cellspacing="0" cellpadding="3">';
-			buff += '<th>Event Name:</th><th>Event Description:</th><th>Event Date:</th>';
-			let events = Object.keys(room.events);
-			for (let i = 0; i < events.length; i++) {
-				buff += `<tr><td>${Chat.escapeHTML(events[i])}</td><td>${room.events[events[i]].desc}</td><td>${Chat.escapeHTML(room.events[events[i]].date)}</td></tr>`;
+
+			let buf = '<table border="1" cellspacing="0" cellpadding="3">';
+			buf += '<th>Event Name:</th><th>Event Description:</th><th>Event Date:</th>';
+			for (let i in events[room]) {
+				buf += `<tr><td>${Chat.escapeHTML(i)}</td><td>${events[room][i].desc}</td><td>${Chat.escapeHTML(events[room][i].date)}</td></tr>`;
 			}
-			buff += '</table>';
-			return this.sendReply(`|raw|<div class="infobox-limited">${buff}</div>`);
+			buf += '</table>';
+			return this.sendReply(`|raw|<div class="infobox-limited">${buf}</div>`);
 		},
 		add: function (target, room, user) {
-			if (room.battle) return this.errorReply("This command is not meant for battle rooms.");
+			if (!room.chatRoomData) return this.errorReply("This command is unavailable in temporary rooms.");
 			if (!this.can('declare', null, room)) return false;
-			if (!room.events) room.events = {};
-			let [eventName, desc, ...date] = target.split('|');
-			date = date.join("|");
-			if (!eventName || !desc || !date) return this.errorReply("You're missing a command parameter - see /help roomevents for this command's syntax.");
+			if (!events[room]) events[room] = {};
+
+			let [eventName, desc, date] = target.split('|').map(param => param.trim());
+			if (!(eventName && desc && date)) return this.errorReply("You're missing a command parameter - see /help roomevents for this command's syntax.");
+
+			if (eventName.length > 50) return this.errorReply("Event names should not exceed 50 characters.");
+			if (date.length > 150) return this.errorReply("Event dates should not exceed 150 characters.");
+			if (desc.length > 500) return this.errorReply("Event descriptions should not exceed 500 characters.");
```

Comment:
I did 500 because that's what /roomfaq does, but I could always bump it up if needed. I intentionally included HTML tags in this limit since I want to encourage people to use HTML sparingly.